### PR TITLE
libusbmuxd: new package

### PR DESCRIPTION
**Pull request by whitfin**
This PR adds the `libusbmuxd` package!

### CODE REVIEW

**Comment by whitfin** [comment]
Note that this is a requirement for using Spack against Google's Flutter toolchain.

File: var/spack/repos/builtin/packages/libusbmuxd/package.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libusbmuxd(Package):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libusbmuxd/package.py]: Looking at the tarball this seems to be an autotools package, thus I think:
```python
class Libusbmuxd(AutotoolsPackage):
```
might be more appropriate. Or is there a reason for not deriving from that base class?

File: var/spack/repos/builtin/packages/libusbmuxd/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    version('1.0.10', sha256='1aa21391265d2284ac3ccb7cf278126d10d354878589905b35e8102104fec9f2')
+    version('1.0.9',  sha256='2e3f708a3df30ad7832d2d2389eeb29f68f4e4488a42a20149cc99f4f9223dfc')
+
+    depends_on('autoconf',   type='build', when='@master')
+    depends_on('automake',   type='build', when='@master')
+    depends_on('libtool',    type='build', when='@master')
+    depends_on('pkg-config', type='build')
+    depends_on('libplist')
+
+    phases = ['autogen', 'install']
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libusbmuxd/package.py]: Usually you don't want to specify `phases` in a `package.py`: this class attribute is better used in base classes to specify a pipeline that all derived packages will follow. If you inherit from `AutotoolsPackage` you'll get the usual:

https://github.com/spack/spack/blob/af4a36c4d99d7953200d369bbeba6303ad87dcf7/lib/spack/spack/build_systems/autotools.py#L52-L53

File: var/spack/repos/builtin/packages/libusbmuxd/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    def autogen(self, spec, prefix):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libusbmuxd/package.py]: It's very likely that this function won't be needed if you inherit from `AutotoolsPackage`

File: var/spack/repos/builtin/packages/libusbmuxd/package.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        if self.spec.satisfies('@master'):
+            autogen = Executable('./autogen.sh')
+            autogen()
+
+    def install(self, spec, prefix):
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/libusbmuxd/package.py]: For `AutotoolsPackage` this function can be substituted by the much simpler:
```python
def configure_args(self):
    return [
        '--disable-dependency-tracking',
        '--disable-silent-rules'
    ]
```

**Comment by whitfin** [comment]
@alalazo I didn't use that package just because of oversight :) I've updated to use it now!

**Pull request merged**